### DISCUSSION TITLE
Riskiness types

### DISCUSSION THREAD

**Discussion by keradus** [PHP-CS-Fixer/PHP-CS-Fixer#7560] [RFC]
We have few types of riskiness, eg
- hardcore like modify `==` -> `===`, for sure will change the meaning of code
- ok if one would not override language level functions in his namespace, like `rand` -> `mt_rand`
- ok if one would not override PHPUnit methods
- annotation manipulation, ok if one not rely on them (eg Doctrine)

Maybe it would be worth to introduce few types of riskiness, allowing for only some of them to happen (like updating phpunit calls is OK, but don't modify `==`/`===`).
For now it's all or nothing.

What do you think ?

**Comment by SpacePossum** [comment]
- casing fixed correctly if one would not override language level functions ([heuristic](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1665))

:+1: 

IMHO it would be nice to combine this with [risky fixer explaination](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1991)

**Comment by SpacePossum** [comment]
This is stalling, yet I think it would still be of value.
It will provide a more fine grain way of using the tool when configuration can be done on the risk-level.
